drivers/flash/Kconfig.simulator

```diff
@@ -0,0 +1,69 @@
+# Kconfig - Flash simulator config
+#
+# Copyright (c) 2018 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+menuconfig FLASH_SIMULATOR
+	bool
+	prompt "Flash simulator"
+	default y if BOARD_QEMU_CORTEX_M3
+	depends on FLASH
+	select STATS
+	select STATS_NAMES
+	select FLASH_HAS_PAGE_LAYOUT
+	select FLASH_HAS_DRIVER_ENABLED
+	help
+	  Enable the flash simulator.
+
+if FLASH_SIMULATOR
+
+config FLASH_SIMULATOR_FLASH_SIZE
```

Comment:
FLASH_SIMULATOR_FLASH_SIZE should be replaced by register size in flash_sim0: node you introduced. (https://github.com/zephyrproject-rtos/zephyr/blob/4952168c14c06e990489de927482155c44665e97/dts/common/sim-flash.dtsi#L15)